examples/light-switch-app/nrfconnect/main/BindingHandler.cpp

```diff
@@ -0,0 +1,199 @@
+    BindingData * data = static_cast<BindingData *>(aContext);
+
+    if (aBinding.type == EMBER_UNICAST_BINDING)
+    {
+        switch (data->ClusterId)
+        {
+        case Clusters::OnOff::Id:
+            OnOffProcessCommandUnicast(data->CommandId, aBinding, aDeviceProxy, aContext);
+            break;
+        case Clusters::LevelControl::Id:
+            LevelControlProcessCommandUnicast(data->CommandId, aBinding, aDeviceProxy, aContext);
+            break;
+        default:
+            LOG_DBG("Invalid binding unicast command data");
+            break;
+        }
+    }
+}
+
+void BindingHandler::InitInternal(intptr_t arg)
+{
+    LOG_INF("Initialize binding Handler");
+    auto & server = Server::GetInstance();
+    if (CHIP_NO_ERROR !=
+        BindingManager::GetInstance().Init(
+            { &server.GetFabricTable(), server.GetCASESessionManager(), &server.GetPersistentStorage() }))
+    {
+        LOG_ERR("BindingHandler::InitInternal failed");
+    }
+
+    BindingManager::GetInstance().RegisterBoundDeviceChangedHandler(LightSwitchChangedHandler);
+    PrintBindingTable();
+}
+
+void BindingHandler::PrintBindingTable()
+{
+    BindingTable & bindingTable = BindingTable::GetInstance();
+
+    LOG_INF("Binding Table [%d]:", bindingTable.Size());
+    uint8_t i = 0;
+    for (auto & entry : bindingTable)
+    {
+        switch (entry.type)
+        {
+        case EMBER_UNICAST_BINDING:
+            LOG_INF("[%d] UNICAST:", i++);
+            LOG_INF("\t\t+ Fabric: %d\n \
+            \t+ LocalEndpoint %d \n \
+            \t+ ClusterId %d \n \
+            \t+ RemoteEndpointId %d \n \
+            \t+ NodeId %d",
+                    (int) entry.fabricIndex, (int) entry.local, (int) entry.clusterId.Value(), (int) entry.remote,
+                    (int) entry.nodeId);
+            break;
+        case EMBER_MULTICAST_BINDING:
+            LOG_INF("[%d] GROUP:", i++);
+            LOG_INF("\t\t+ Fabric: %d\n \
+            \t+ LocalEndpoint %d \n \
+            \t+ RemoteEndpointId %d \n \
+            \t+ GroupId %d",
+                    (int) entry.fabricIndex, (int) entry.local, (int) entry.remote, (int) entry.groupId);
+            break;
+        case EMBER_UNUSED_BINDING:
+            LOG_INF("[%d] UNUSED", i++);
+            break;
+        case EMBER_MANY_TO_ONE_BINDING:
+            LOG_INF("[%d] MANY TO ONE", i++);
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+void BindingHandler::SwitchWorkerHandler(intptr_t context)
+{
+    VerifyOrReturn(context != 0, LOG_ERR("Invalid Swich data"));
+
+    BindingData * data = reinterpret_cast<BindingData *>(context);
+    LOG_INF("Notify Bounded Cluster | endpoint: %d cluster: %d", data->EndpointId, data->ClusterId);
+    BindingManager::GetInstance().NotifyBoundClusterChanged(data->EndpointId, data->ClusterId, static_cast<void *>(data));
+
+    Platform::Delete(data);
```

Comment:
I am just wondering who is an owner of incoming context? I mean, it looks like we are deliberately deleting data hooked by `context`, it rather violates RAII approach. @Damian-Nordic, @kkasperczyk-no are we allowed to use smart pointers in Matter? At least the `std::uniqe_ptr` should not cause any additional overhead.

Reply:
After calling Notifying Cluster, data hooked by context is no longer needed so that we can free this memory.

Reply:
We are allowed and we have our own wrapper: `MakeUnique` for that. But in this case it's tricky because the common interface just takes a raw pointer.